### ISSUE TITLE
[no-misused-promises] allow using async functions in event listeners in React

### ISSUE THREAD

**Issue by Mikilll94** [typescript-eslint/typescript-eslint#11008]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Issue Description

![no-misused-promises rule in failing in VSCode editor](https://github.com/user-attachments/assets/1b472cba-a787-4bfd-8ac9-3732d51069ba)

Rule `no-misused-promises` reports false positive in this case. It is perfectly fine to use async function directly in event handlers in React and I think that most devs are doing that.

### Reproduction Repository Link

https://github.com/Mikilll94/no-misuses-promises-repro

### Repro Steps

1) clone the repo
2) `npm install`
3) `npm run lint`

### Versions

├── @eslint/js@9.23.0
├── @types/react-dom@19.0.4
├── @types/react@19.0.12
├── @vitejs/plugin-react@4.3.4
├── eslint-plugin-react-hooks@5.2.0
├── eslint-plugin-react-refresh@0.4.19
├── eslint@9.23.0
├── globals@15.15.0
├── react-dom@19.1.0
├── react@19.1.0
├── typescript-eslint@8.28.0
├── typescript@5.7.3
└── vite@6.2.3

**Comment by bradzacher** [comment]
> It is perfectly fine to use async function directly in event handlers in React

Why do you say it's perfectly fine?
Does react have special handling for async event handlers?
AFAIK it doesn't - which means your async functions will create floating promises and unhandled rejections.

**Comment by bradzacher** [comment]
Something like
```ts
onClick={() => {
  doSomethingAsync()
   .catch(handleError)
}}
```

Or if you believe your code is right, use a disable comment.
